include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -128,27 +129,130 @@ namespace LinearAlgebra
         resize_val(const types::global_dof_index new_alloc_size,
                    types::global_dof_index &     allocated_size,
                    ::dealii::MemorySpace::
-                     MemorySpaceData<Number, ::dealii::MemorySpace::Host> &data)
+                     MemorySpaceData<Number, ::dealii::MemorySpace::Host> &data,
+                   const MPI_Comm &comm_shared)
         {
-          if (new_alloc_size > allocated_size)
+          if (comm_shared == MPI_COMM_SELF)
```

Comment:
Is this check enough or do we need `MPI_Comm_compare` and check for `MPI_IDENT` or `MPI_CONGRUENT`? It probably suffices with `==`.

Reply:
I have addressed the above comments. I am not sure about this one: I can test later of `MPI_COMM_SELF` is identical to a subgroup of size 1. We could tackle this possible issue in a follow-up PR.